packages/core/src/app/shipping/ShippingAddress.tsx

```diff
@@ -81,31 +88,44 @@ const ShippingAddress: FunctionComponent<ShippingAddressProps> = (props) => {
         onFieldChange(fieldName, value);
     };
 
-    if (methodId) {
-        let options: ShippingInitializeOptions = {};
+    if (methodId === 'amazonpay' && shippingAddress) {
```

Comment:
Can we please refactor this logic to a separate component, maybe a component factory?

Reply:
It might be better to do that in separate PR, because we should stick to the goal of current implementation. Also we should think about moving this code to a separate package (like for Payment components), but we don't have such mechanism for Shipping for now. 

Lets put this topic in a list for future discussions

Reply:
I've created a separate PR for this:
https://github.com/bigcommerce/checkout-js/pull/1865